### PR TITLE
RELATED: BB-257 onFiredDrillEvent

### PR DESCRIPTION
**Pull request by BugsBunny338**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1/ : SUCCESS in 51s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/634/ : SUCCESS in 2m 08s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/2/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/635/ : FAILURE in 2m 55s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/3/ : SUCCESS in 38s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/636/ : FAILURE in 1m 18s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/9/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/643/ : SUCCESS in 1m 57s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/10/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/644/ : SUCCESS in 1m 55s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/11/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/645/ : SUCCESS in 48s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/12/ : SUCCESS in 38s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/646/ : SUCCESS in 1m 53s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/13/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/647/ : SUCCESS in 50s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/14/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/648/ : SUCCESS in 1m 51s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/15/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/649/ : SUCCESS in 1m 54s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/16/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/650/ : SUCCESS in 51s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/17/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/651/ : SUCCESS in 53s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/18/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/652/ : FAILURE in 21s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/19/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/653/ : SUCCESS in 1m 40s

File: src/interfaces/DrillEvents.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { Afm } from '@gooddata/data-layer';
+
+export interface IDrillEvent {
+    executionContext: Afm.IAfm;
+    drillContext: {
+        type: 'pie' | 'column' | 'bar' | 'line' | 'table';
```
Comment (martinmosko) [inline src/interfaces/DrillEvents.ts]: what about enums ? 

```
enum E {
    hello = "hello",
    world = "world"
};
```

File: src/interfaces/DrillEvents.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        element: 'slice' | 'bar' | 'point' | 'cell';
+        x?: number;
+        y?: number;
+        columnIndex?: number;
+        rowIndex?: number;
+        row?: any[];
```
Comment (martinmosko) [inline src/interfaces/DrillEvents.ts]: we do not know the row structure?

File: src/interfaces/DrillEvents.ts (hunk 1 above)
Comment (BugsBunny338) [reply src/interfaces/DrillEvents.ts]: Good point, how about reusing https://github.com/gooddata/gooddata-react-components/blob/master/src/components/afm/SimpleDataAdapterProvider.tsx#L21?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/23/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/658/ : SUCCESS in 1m 59s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/25/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/660/ : SUCCESS in 1m 52s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/26/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/661/ : SUCCESS in 1m 08s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-merge-compliance-check/67/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/662/ : SUCCESS in 1m 54s

**Pull request merged**